templates/contribute/doc.md

```diff
@@ -253,6 +254,51 @@ where we give overviews spanning several Lean files, and
 more mathematical explanations in cases where formalization requires slightly exotic points of view,
 see for instance the [topology documentation](../theories/topology.html).
 
+## Citing other works
+
+To cite papers and books in doc strings, the references should first be added
+to the BibTeX file: `docs/references.bib`. To normalize the file with `bibtool`, you
+can run:
+
+```text
+bibtool --preserve.key.case=on --preserve.keys=on --print.use.tab=off -s -i docs/references.bib -o docs/references.bib
+```
+
+To ensure that your citations become links in the online docs, you can use either of the
+following two styles:
+
+First, you may enclose the citation key used in `docs/references.bib` in square brackets:
+
+```markdown
+The proof can be found in [Boole1854].
```

Comment:
None of the three examples in this section exist in my `docs/references.bib`.  And they use different styles.

More seriously, I think we need a naming convention for the bibliography keys.

Reply:
Bibtool can normalize the keys with very flexible formatting options.

Reply:
If we remove `--preserve.key.case` and `--preserve.keys` and add `-f short` it formats to this:
```diff
-
-@Article{         ahrens2017,
+@Article{         ahrens.lumsdaine:displayed,
   author        = {Benedikt Ahrens and Peter LeFanu Lumsdaine},
   year          = {2019},
   title         = {Displayed Categories},
@@ -11,7 +11,7 @@
   doi           = {10.23638/LMCS-15(1:20)2019}
 }
 
-@Book{            aluffi2016,
+@Book{            aluffi:algebra,
   title         = {Algebra: Chapter 0},
   author        = {Aluffi, Paolo},
   series        = {Graduate Studies in Mathematics},
@@ -22,7 +22,7 @@
                   Society}
 }
 
-@Book{            atiyah-macdonald,
+@Book{            atiyah.macdonald:introduction,
   author        = {Atiyah, M. F. and Macdonald, I. G.},
   title         = {Introduction to commutative algebra},
   publisher     = {Addison-Wesley Publishing Co., Reading, Mass.-London-Don
@@ -34,7 +34,7 @@
   mrreviewer    = {J. A. Johnson}
 }
```

But of course this will break citations already in the library. I can imagine people editing the bib, writing comments, formatting the bib, and forgetting to update the new comments.